onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py

```diff
@@ -141,7 +141,7 @@ def chain_model(args):
 
     # Initializers/opsets
     # Delete shared data between decoder/encoder and move to larger graph initializers
-    initializers = get_shared_initializers(encoder_model, decoder_model)
+    initializers = get_shared_initializers(decoder_model, encoder_model, require_raw_data=True)
```

Comment:
I think the parameter order should be `(encoder_model, decoder_model, require_raw_data=True)` to match the function definition and to add the correct prefix to the names.
https://github.com/microsoft/onnxruntime/blob/85ad589c6f088a66b80ec864f59db4993a0b3ab8/onnxruntime/python/tools/transformers/convert_generation.py#L990-L998

Reply:
This is true - and good observation with the prefixes. The reason I switched these around is because of how the loops are written in the deduplication process:

for remove_shared_initializers(graph A, graph B), the method will loop through every initializer in graph A, and see if they exist in graph B.

The decoder graph has most initializers shared with the encoderdecoderinit, while the encoderdecoderinit has fewer (half?) of initializers shared with encoderdecoderinit. 

From some basic testing, setting the decoder as graph A causes the loop to run fewer times (fewer initializers? randomness?) and caused the deduplication process to run about 10% faster. You're still right that the prefixes won't match up, but I think the speedup justifies the strangeness for the moment.

(I also think rewriting the code path to fix the prefixes in this case is a bit out of scope for this PR)